app/policies/theblog/content_node_policy.rb

```diff
@@ -25,7 +25,7 @@ def unblock?
     end
 
     def account_can_update_record?
-      (user == record.author && account_has_role?(user, [:editor])) ||
+      (user.try(:id) == record.author.try(:id) && account_has_role?(user, [:editor])) ||
```

Comment:
once I've introduced Theblog::Account, user and record author are instances of different models, Theblog::Account and Incarnator.account_model. So it's only way to compare them by their ids